### PR TITLE
docs: move fabric on its own

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

a suggestion to have Fabric in the unified docs instead of PL
cc: @awaelchli 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @carmocca @akihironitta @justusschock @awaelchli

### CODE REVIEW

File: requirements/fabric/docs.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-r ../docs.txt
+
+lai-sphinx-theme
+tensorboardX>=2.2, <=2.5.1  # min version is set by torch.onnx missing attribute
```
Comment (lantiga) [inline requirements/fabric/docs.txt:4]: do we need tensorboard for docs?

File: docs/source-fabric/index.rst [hunk 2]
```diff
@@ -60,20 +60,20 @@ Fabric is the fast and lightweight way to scale PyTorch models without boilerpla
 Why Fabric?
 ***********
 
-Fabric differentiates itself from a fully-fledged trainer like :doc:`Lightning Trainer <../common/trainer>` in these key aspects:
+Fabric differentiates itself from a fully-fledged trainer like Lightning Trainer in these key aspects:
```
Comment (lantiga) [inline docs/source-fabric/index.rst:63]: Can't we have links between top level projects?

File: docs/source-fabric/guide/lightning_module.rst [hunk 3]
```diff
@@ -7,7 +7,7 @@ Organize Your Code
 Any raw PyTorch can be converted to Fabric with zero refactoring required, giving maximum flexibility in how you want to organize your projects.
 
 However, when developing a project in a team or sharing the code publicly, it can be beneficial to conform to a standard format of how core pieces of the code are organized.
-This is what the :doc:`LightningModule <../../common/lightning_module>` was made for!
+This is what the LightningModule was made for!
```
Comment (lantiga) [inline docs/source-fabric/guide/lightning_module.rst:10]: Can't we have links between top-level projects?

**Review by lantiga — CHANGES_REQUESTED** [review]
We should preserve links between Fabric and PyTorch Lightning

File: docs/source-fabric/index.rst (hunk 2 above)
Comment (Borda) [reply docs/source-fabric/index.rst:63]: Not as they are generated as very independent documentation; we may later explore https://docs.readthedocs.io/en/stable/subprojects.html (which we tried already in past/a few months ago, but did not work well)

File: docs/source-fabric/guide/lightning_module.rst (hunk 3 above)
Comment (Borda) [reply docs/source-fabric/guide/lightning_module.rst:10]: We can have later html ref, but note that we are generating each dos as stand-alone no jointly, see laos: https://github.com/Lightning-AI/lightning/pull/16742#discussion_r1119790964

File: requirements/fabric/docs.txt (hunk 1 above)
Comment (Borda) [reply requirements/fabric/docs.txt:4]: just refering what is there, but I am fine to switch it :)

File: docs/source-fabric/guide/lightning_module.rst (hunk 3 above)
Comment (awaelchli) [reply docs/source-fabric/guide/lightning_module.rst:10]: I suggest we put the https url here to lightning stable docs

**Review by awaelchli — APPROVED** [review]
LGTM
If we could replace the previous links to PL with actual URLs, thanks!

**Comment by justusschock** [comment]
since they are all standalone sphinx docs, we should be able to use [intersphinx mapping] pointing to the indices of each project and then have links like 
```
:class:`pytorch_lightning.core.module.LightningModule`
```
automatically resolved.

File: docs/source-fabric/guide/lightning_module.rst (hunk 3 above)
Comment (Borda) [reply docs/source-fabric/guide/lightning_module.rst:10]: spoke with @justusschock and we bay link external docs similar as we refer to PyTorch or NumPy... :otter:

Comment (Borda) [reply docs/source-fabric/guide/lightning_module.rst:10]:
```suggestion
This is what the :class:`pytorch_lightning.core.module.LightningModule` was made for!
```

File: docs/source-fabric/guide/lightning_module.rst [hunk 4]
```diff
@@ -62,7 +62,7 @@ Take these main ingredients and put them in a LightningModule:
             ...
 
 
-This is a minimal LightningModule, but there are :doc:`many other useful hooks <../../common/lightning_module>` you can use.
+This is a minimal LightningModule, but there are many other useful hooks you can use.
```
Comment (Borda) [inline docs/source-fabric/guide/lightning_module.rst:65]:
```suggestion
This is a minimal :class:`pytorch_lightning.core.module.LightningModule`, but there are `many other useful hooks <https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#hooks>`_ you can use.
```

File: docs/source-fabric/index.rst (hunk 2 above)
Comment (Borda) [reply docs/source-fabric/index.rst:63]:
```suggestion
Fabric differentiates itself from a fully-fledged trainer like Lightning :class:`pytorch_lightning.Trainer` in these key aspects:
```

File: docs/source-fabric/index.rst [hunk 5]
```diff
@@ -60,20 +60,20 @@ Fabric is the fast and lightweight way to scale PyTorch models without boilerpla
...
 
 **Fast to implement**
 There is no need to restructure your code: Just change a few lines in the PyTorch script and you'll be able to leverage Fabric features.
 
 **Maximum Flexibility**
 Write your own training and/or inference logic down to the individual optimizer calls.
-You aren't forced to conform to a standardized epoch-based training loop like the one in :doc:`Lightning Trainer <../common/trainer>`.
+You aren't forced to conform to a standardized epoch-based training loop like the one in Lightning Trainer.
```
Comment (Borda) [inline docs/source-fabric/index.rst:70]:
```suggestion
You aren't forced to conform to a standardized epoch-based training loop like the one in Lightning :class:`pytorch_lightning.Trainer`.
```

File: docs/source-fabric/index.rst [hunk 6]
```diff
@@ -60,20 +60,20 @@ Fabric is the fast and lightweight way to scale PyTorch models without boilerpla
...
 You can do flexible iteration based training, meta-learning, cross-validation and other types of optimization algorithms without digging into framework internals.
 This also makes it super easy to adopt Fabric in existing PyTorch projects to speed-up and scale your models without the compromise on large refactors.
 Just remember: With great power comes a great responsibility.
 
 **Maximum Control**
-The :doc:`Lightning Trainer <../common/trainer>` has many built in features to make research simpler with less boilerplate, but debugging it requires some familiarity with the framework internals.
+The Lightning Trainer has many built in features to make research simpler with less boilerplate, but debugging it requires some familiarity with the framework internals.
```
Comment (Borda) [inline docs/source-fabric/index.rst:76]:
```suggestion
The Lightning :class:`pytorch_lightning.Trainer` has many built-in features to make research simpler with less boilerplate, but debugging it requires some familiarity with the framework internals.
```

File: requirements/fabric/docs.txt (hunk 1 above)
Comment (Borda) [reply requirements/fabric/docs.txt:4]:
```suggestion
tensorboard
```

**Comment by Borda** [comment]
> We should preserve links between Fabric and PyTorch Lightning

@lantiga replaced links to API and living docs... mind rechecking it? :otter:

**Comment by Borda** [comment]
results in https://github.com/Lightning-AI/lightning/actions/runs/4297938862

**Review by lantiga — APPROVED** [review]
LGTM!

**Review by akihironitta — APPROVED** [review]

File: docs/source-pytorch/conf.py [hunk 7]
```diff
@@ -74,7 +74,7 @@ def _transform_changelog(path_in: str, path_out: str) -> None:
     shutil.copy(md, os.path.join(PATH_HERE, FOLDER_GENERATED, os.path.basename(md)))
 # copy also the changelog
 _transform_changelog(
-    os.path.join(PATH_ROOT, "src", "lightning", "pytorch", "CHANGELOG.md"),
+    os.path.join(PATH_ROOT, "src", "lightning", "fabric", "CHANGELOG.md"),
```
Comment (carmocca) [inline docs/source-pytorch/conf.py:77]: Why this change?

**Review by carmocca — COMMENTED** [review]
This PR is now blocking master. It introduced dead links that are reference in our markdown files, so the "markdown-link-checker" bot is failing. Example: https://github.com/Lightning-AI/lightning/actions/runs/4303549950/jobs/7503422165
